### ISSUE TITLE
Stray (-dx12, UE4) crashing within a minute or two of gameplay

### ISSUE THREAD

**Issue by kode54** [HansKristian-Work/vkd3d-proton#2078]
Please describe your issue as accurately as possible. Include screenshots or videos if relevant

### Software information
Stray (Steam AppID 1332010)
Running with -dx12 startup switch

### System information
- GPU: AMD RX 6700 XT
- Driver: Mesa (24.1.6 from Arch repos, also 24.3.0_devel.193808.4f1c898bb33 built with mesa-tkg from gitlab.freedesktop.org/pixelcluster/mesa vm-explicit-sync branch)
- Wine version: GE-Proton9-11
- VKD3D-Proton version: As bundled with GE-Proton9-11

### Log files
[steam-1332010.log](https://github.com/user-attachments/files/16719371/steam-1332010.log)

### Notes
Happens with both Mesa and amdvlk, so it's likely either a game bug or vkd3d bug.

**Comment by kode54** [comment]
Yes, it does indeed appear to be ray tracing related. So whose bug is this now, Mesa's? Or still the game's for having buggy (and undocumented) ray tracing code?

**Comment by RareMv** [comment]
A lot of UE4 games, built on 4.19 or newer, can use -dx12 as a launch command to run in DX12 instead of DX11. To my knowledge, Stray, as a normally DX11 only UE4 title, is the only game where running this launch command also triggers RT effects, assuming the hardware/software is RT capable.

Now, this is clearly a design quirk. But RT effects get triggered and the game doesn't crash on Windows on RT capable GPUs. So it shouldn't crash on Linux either, given DXR is now enabled by default on Mesa.

There was some talks where either Mesa or VKD3D was meant to blacklist RT in Stray in DX12 due to Steam Deck in particular, but I'm not sure how that discussion went.

**Comment by shelterx** [comment]
Mhhm, didn't know RT in Stray was a thing. Seems to work with Nvidia so perhaps an AMD mesa bug?
